cumulusci/tasks/bulkdata/step.py

```diff
@@ -255,7 +251,14 @@ def get_results(self):
         def convert(rec):
             return [str(rec[f]) if rec[f] is not None else "" for f in self.fields]
 
-        yield from (convert(rec) for rec in self.response["records"])
+        while True:
+            yield from (convert(rec) for rec in self.response["records"])
+            if not self.response["done"]:
+                self.response = self.sf.query_more(
+                    self.response["nextRecordsUrl"], identifier_is_url=True
+                )
+            else:
+                return
```

Comment:
```suggestion
        while not self.response["done"]:
            yield from (convert(rec) for rec in self.response["records"])
            self.response = self.sf.query_more(
                self.response["nextRecordsUrl"], identifier_is_url=True
            )
```

Reply:
This would exit early if the query returns a number of results 0<n<batch size, wouldn't it?

Reply:
Yep. I thought it felt too good to be true.